canvas_modules/common-canvas/src/tooltip/tooltip.jsx

```diff
@@ -331,6 +332,8 @@ class ToolTip extends React.Component {
 	}
 
 	render() {
+		const inputTruncated = this.triggerRef && this.triggerRef.getElementsByTagName("input")[0];
+		this.truncatedElement = !(inputTruncated && inputTruncated.scrollWidth > inputTruncated.clientWidth);
```

Comment:
Hi @nabilIbm, this code right here is what's calculating if the Textfield should be truncated or not. This logic should be moved to `TruncatedContentTooltip` instead. Then no changes are required in `Tooltip`. Inside `TruncatedContentTooltip`, if you determine that the `content` needs to be truncated, you will return the `<Tooltip>`, otherwise, just return the `content`